lib/sinatra/base.rb

```diff
@@ -1522,8 +1522,8 @@ def prototype
       # Create a new instance of the class fronted by its middleware
       # pipeline. The object is guaranteed to respond to #call but may not be
       # an instance of the class new was called on.
-      def new(*args, &bk)
-        instance = new!(*args, &bk)
+      def new(*args, **kwargs, &bk)
+        instance = kwargs.length > 0 ? new!(**kwargs, &bk) : new!(*args, &bk)
```

Comment:
Out of curiosity, what happens if we just did this:

```
instance = new!(*args, **kwargs, &bk)
```

I think the `initialize` method signature would need updated, but would that work? Or would that break something?

Reply:
hey @jkowens 
thanks for the suggestion!

It worked perfectly!

already pushed the changes

Reply:
I think updating the `initialize` method signature will resolve some of the build failures: https://github.com/sinatra/sinatra/blob/cb64dd3516e486f79b3ecaad2c473830d61e56b5/lib/sinatra/base.rb#L919

```
def initialize(app = nil, **kwargs)
```

Reply:
hey @jkowens, now build is breaking because of 
```
Template engine not found: wlang
```

Reply:
Hmm that one doesn't make sense to me!

Reply:
Ah! It looks like wlang 3.0 isn't compatible at the the moment. Can you go to the Gemfile and lock wlang to '~> 2.3.1'?

Reply:
Actually, I'll do that in a separate commit 👍

Reply:
Please rebase master and I think it should be good to go!